### PR TITLE
Add optional config field to tool step

### LINKED ISSUE #878: [FEATURE] Ml-commons plugin introducing a new optional field "config" in MLTooSpec
### Is your feature request related to a problem?
Ml-commons is introducing a new field `config` in `MLToolSpec` as showed in the pr https://github.com/opensearch-project/ml-commons/pull/2977. With the change, we allow cx to configure some static parameters for tools that stay unchanged/persist during the tool execution. It will fix a couple of bugs mentioned in the pr.

### What solution would you like?
The pr https://github.com/opensearch-project/ml-commons/pull/2977.

### What alternatives have you considered?
N/A.

### Do you have any additional context?
It's an optional field.

### PR DESCRIPTION
**Pull request by dbwiddis**
### Description

Adds the optional `config` map to the `ToolStep`

### Related Issues

Resolves #878 

Draft, pending merge:
 - https://github.com/opensearch-project/ml-commons/pull/2977

### Check List
- [x] New functionality includes testing.
- [x] New functionality has been documented.
- [x] Commits are signed per the DCO using `--signoff`.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/flow-framework/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by amitgalitz — APPROVED** [review]
Looks good, but as mentioned should be merged only after ml commons PR is merged and we can then uncomment the lines in ToolStep and ToolStepTests

**Review by joshpalis — APPROVED** [review]
lgtm, approved to unblock

**Pull request merged**
